### PR TITLE
Enumerable Methods

### PR DESCRIPTION
**Pull request by ioanniskousis**
This project implements my own version of particular Ruby’s enumerable methods.
Includes enumerable.rb, the main file where Enumerable module is hosted, and several testing files.
Tests can be applied for each method independently running the corresponding file from tests folder or all-together at once running the tests.rb file on the root

### CODE REVIEW

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      my_each { |x| return false if (x =~ args[0]).nil? }
+      return true
+    end
+    unless block_given?
+      my_each { |x| return false if x.nil? || x == false }
+      return true
+    end
+    my_each { |x| return false unless yield(x) }
+    true
+  end
```
Comment (majovanilla) [inline enumerable.rb]: You are not considering the option of giving a class or a number as an argument. For example:
`some_enumerable.my_all?(String)`
`some_enumerable.my_all?(6)`
Right now, those examples return false on truthy cases.

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      my_each { |x| return true if x =~ args[0] }
+      return false
+    end
+    unless block_given?
+      my_each { |x| return true if !x.nil? && (x != false) }
+      return false
+    end
+    my_each { |x| return true if yield(x) }
+    false
+  end
```
Comment (majovanilla) [inline enumerable.rb]: You are not considering the option of giving a class as an argument. For example:
`some_enumerable.my_any?(String)`

Also, your method raises an error when passing a string. For example:
`some_enumerable.my_any?('cow)`

File: enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    end
+    unless block_given?
+      my_each { |x| return false unless x.nil? || (x == false) }
+      return true
+    end
+    my_each do |x|
+      return false if yield(x)
+    end
+    true
+  end
```
Comment (majovanilla) [inline enumerable.rb]: You are not considering the option of giving a class or a number as an argument. For example:
`some_enumerable.my_none?(String)`
`some_enumerable.my_none?(6)`
Right now, those examples return false on truthy cases.

**Review by majovanilla — CHANGES_REQUESTED** [review]
Hey @ioanniskousis , 👋

## You did an *amazing*🌟 ✨ job on this project! 👊🏻  💯

- Good Readme file
- Good Pull Request Message
- Almost all your methods pass the tests. And that's quite an accomplishment!! 

There are still some changes 👇 you need to implement to approve this request 🔍
Please check the commented code below.

## Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 


Happy coding 🤓

--Maria Reyes

File: enumerable.rb (hunk 3 above)
Comment (ioanniskousis) [reply enumerable.rb]: I improved my_none method to support Class and object as argument

File: enumerable.rb (hunk 2 above)
Comment (ioanniskousis) [reply enumerable.rb]: I improved my_any method to support Class and object as argument

File: enumerable.rb (hunk 1 above)
Comment (ioanniskousis) [reply enumerable.rb]: I improved my_all method to support Class and object as argument

**Comment by ioanniskousis** [comment]
1. I improved my_all method to support Class and object as argument
2. I improved my_any method to support Class and object as argument
3. I improved my_none method to support Class and object as argument

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hi @ioanniskousis, you have done an amazing work:+1:. Just an error left to be fixed now:**
- When a regex is passed to **my_any?**, it should return true if there is at least one of the collection that matches the regex. For example
`%w[dog door rod blade].my_any?(/d/)` should return `true`


_**Kindly fix this and submit another code review request.**_

**Any question? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Comment by ioanniskousis** [comment]
> **Hi @ioanniskousis, you have done an amazing work👍. Just an error left to be fixed now:**
> 
> * When a regex is passed to **my_any?**, it should return true if there is at least one of the collection that matches the regex. For example
>   `%w[dog door rod blade].my_any?(/d/)` should return `true`
> 
> _**Kindly fix this and submit another code review request.**_
> 
> **Any question? You can reach me on Slack (Moses Ogwo)**
> 
> **Cheers🌠**
> 
> ### Status: Changes Requested.

I rearranged the check for a Regexp with the check for including an object and it worked

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Nice Progress :+1: 
You have come a long way with this project. Your code covers most edge cases but we still have one failing edge case to fix before we can move forward.
- [x] `my_any?` when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
- [x] A sample edge case. Use this edge case to guide you in debugging the code.
```ruby
words = %w[dog door rod blade] 
words.my_any?('cat') === words.any?('cat') # should return true because none of the collection matches the pattern.
```

### Martin Afani

**Comment by ioanniskousis** [comment]
Mr Martin Afani,
are you sure you made a correct test ?
Because : %w[dog door rod blade].my_any?('cat') returns FALSE
                 %w[dog door rod blade].any?('cat') returns FALSE
AND       %w[dog door rod blade].my_any?('cat') === %w[dog door rod blade].any?('cat')
RETURNS TRUE

WHAT EXACTLY IS THE ERROR ?

**Comment by majovanilla** [comment]
Hey @ioanniskousis , 👋

## You did an *amazing*🌟 ✨ job on this project! 👊🏻  💯

## Your methods are working as expected! Congratulations!! 😀

### Your project is **APPROVED**, you can merge your branch and submit the project completion form.


If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 

Happy coding 🤓

--Maria Reyes

**Comment by ioanniskousis** [comment]
Thank you Maria

**Pull request merged**